google-cloud-spanner/lib/google/cloud/spanner/results.rb

```diff
@@ -26,7 +26,12 @@ class Results
         ##
         # Indicates the field names and types for the rows in the returned data.
         #
-        # @return [Hash] The types of the returned data.
+        # @param [Boolean] pairs Allow the types to be represented as a nested
+        #   Array of pairs rather than a Hash. This is useful when results have
+        #   duplicate names. The default is `false`.
```

Comment:
We should error out if this if false and we encounter duplicate names or multiple fields with no names.